.github/workflows/deploy.yml

```diff
@@ -0,0 +1,71 @@
+name: Build and Deploy to S3
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - development
+      - base
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    environment:
+      name: development ##Replace with actual environment after testing
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Use Node.js ${{ matrix.node_version }}
+        uses: actions/setup-node@v4.0.2
+        with:
+          node-version: 'lts/*'
+
+      - name: Cache dependencies
+        uses: actions/cache@v3
+        id: yarn-cache
+        with:
+          path: ./node_modules
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+
+      - name: Install dependencies
+        run: yarn install
+
+      - name: Export secrets to environment variables
+        uses: oNaiPs/secrets-to-env-action@v1.5
+        with:
+          secrets: ${{ toJSON(secrets) }}
+
+      - name: Substitute environment variables
+        uses: tvarohohlavy/inplace-envsubst-action@v1.0.0
+        with:
+          files: .env.deploy
+
+      - name: Replace env file
+        run: mv .env.deploy .env
+
+      - name: Build Docker
+        run: docker build .
```

Comment:
I think for DIMES this is actually probably part of tests, since we're not actually doing anything with this image, we just want to make sure it builds.